android/src/main/cpp/NativeProxy.cpp

```diff
@@ -230,20 +230,64 @@ void NativeProxy::installJSIBindings(
 
   // Layout Animations Start
 
-  auto notifyAboutProgress = [=](int tag, jsi::Value progress) {
-    this->layoutAnimations->cthis()->notifyAboutProgress(progress, tag);
+  auto progressLayoutAnimation = [this](int tag, jsi::Value progress) {
+    this->layoutAnimations->cthis()->progressLayoutAnimation(tag, progress);
   };
 
-  auto notifyAboutEnd = [=](int tag, bool isCancelled) {
-    this->layoutAnimations->cthis()->notifyAboutEnd(tag, (isCancelled) ? 1 : 0);
+  auto endLayoutAnimation = [this](int tag, bool isCancelled, bool removeView) {
+    this->layoutAnimations->cthis()->endLayoutAnimation(
+        tag, isCancelled, removeView);
   };
 
   std::shared_ptr<LayoutAnimationsProxy> layoutAnimationsProxy =
       std::make_shared<LayoutAnimationsProxy>(
-          notifyAboutProgress, notifyAboutEnd);
+          progressLayoutAnimation, endLayoutAnimation, errorHandler);
   std::weak_ptr<jsi::Runtime> wrt = animatedRuntime;
+  std::weak_ptr<LayoutAnimationsProxy> weakLayoutAnimationsProxy =
+      layoutAnimationsProxy;
   layoutAnimations->cthis()->setWeakUIRuntime(wrt);
 
+  layoutAnimations->cthis()->setAnimationStartingBlock(
+      [wrt, weakLayoutAnimationsProxy](
+          int tag,
+          alias_ref<JString> type,
+          alias_ref<JMap<jstring, jstring>> values) {
+        auto rt = wrt.lock();
+        auto layoutAnimationsProxy = weakLayoutAnimationsProxy.lock();
+        if (!rt || !layoutAnimationsProxy) {
+          return;
+        }
+        jsi::Object yogaValues(*rt);
```

Comment:
I know it's a detail, but in our codebase `rt` usually represents a reference to runtime (`jsi::Runtime &`), not a shared pointer (`std::shared_ptr<jsi::Runtime>`).

```suggestion
        auto runtime = wrt.lock();
        auto layoutAnimationsProxy = weakLayoutAnimationsProxy.lock();
        if (!runtime || !layoutAnimationsProxy) {
          return;
        }
        jsi::Runtime &rt = *runtime;
        jsi::Object yogaValues(rt);
```